cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -177,17 +178,24 @@ func Run(s *options.CMServer) error {
 		return err
 	}
 
-	leaderelection.RunOrDie(leaderelection.LeaderElectionConfig{
+	// TODO: enable other lock types
+	rl := resourcelock.EndpointsLock{
 		EndpointsMeta: api.ObjectMeta{
 			Namespace: "kube-system",
 			Name:      "kube-controller-manager",
 		},
-		EndpointsClient: kubeClient,
-		Identity:        id,
-		EventRecorder:   recorder,
-		LeaseDuration:   s.LeaderElection.LeaseDuration.Duration,
-		RenewDeadline:   s.LeaderElection.RenewDeadline.Duration,
-		RetryPeriod:     s.LeaderElection.RetryPeriod.Duration,
+		Client: kubeClient,
```

Comment:
Since you are already changing it, we should do use ClientSet, instead of kubeClient.

I've already sent out a PR that is slightly improving it:
https://github.com/kubernetes/kubernetes/pull/33474

Those two will be fighting with collisions, but I think both are needed.